lib/structures/VoiceState.js

```diff
@@ -31,6 +32,7 @@ class VoiceState extends Base {
         this.suppress = data.suppress !== undefined ? data.suppress : this.suppress || false; // Bots ignore this
         this.selfMute = data.self_mute !== undefined ? data.self_mute : this.selfMute || false;
         this.selfDeaf = data.self_deaf !== undefined ? data.self_deaf : this.selfDeaf || false;
+        this.selfStream = data.self_stream || false;
```

Comment:
Copy the undefined check from above

Reply:
@abalabahaha Sadly the way that discord handles the `self_stream` property is different from the others. Initially, I used the undefined check above it, and during testing discovered that it was broken. After much debugging I realized that discord doesn't pass the prop when the user goes offline so it was never

```js
this.selfStream = data.self_stream !== undefined ? data.self_stream : this.selfStream || false;
```

So basically discord does NOT pass in the `self_stream` when the user is offline. Aka it will be undefined. When it is `undefined` it resolves to `this.selfStream` (which is currently true) `|| false`. So it always remained true

Once they become true it stayed true, so I had to remove the undefined check
if discord passes that value they just went live. If they don't pass it the user went offline